### ISSUE TITLE
Validate elkt graph

### ISSUE THREAD

**Issue by Baltoli** [eclipse/elk#715]
Is there a tool (other than ElkLive) I can use to validate a graph in ELKt format (i.e. looking for syntax / structural errors)?

I have an input graph that ElkLive will not render (https://gist.github.com/Baltoli/6e173a4d838f5a83725feba820d0edc9), but I can't tell why. When converted to JSON, some of the edge sources and targets go missing - as far as I can see, the edges exist in the source graph but are lost when going to JSON.

I'm basically looking to see what's wrong with my input graph (as I presume that's where the problem is, as ElkLive won't render it). The JSON I get after converting is here https://gist.github.com/Baltoli/0c07527b7e105b7807f7c83ff62ac448.

As always thanks for your help - really appreciate it!

**Comment by uruuru** [comment]
The `elkt` format is based on Xtext. As such, Xtext's validation mechanisms could be used. 

Pasting your example into the corresponding editor in Eclipse yields the following problems:

![image](https://user-images.githubusercontent.com/7552459/95115557-f76a2a80-0745-11eb-9daa-2f67d4236cdf.png)

From a first glance at it it looks like 
a) there's a structural problem in that `}` are missing or 
b) that some node/port identifiers do not match to the ones used in edges
c) the containment of edges is not as expected (the warnings; I would expect the graph to be rendered in this case though). 


Three further notes:
1) elk-live's elkt editor should, in principle, provide the same validation results as the Eclipse-based editor. For some reason your graph seems to be hard to parse for Xtext, thus takes a couple of seconds before finishing. I could imagine elk-live timing out.
2) Xtext's validation can be called directily without having an editor. We don't offer any frontend for this, though.
3) What I mentioned relates to validation of the `elkt` format (parsing), not the internal graph representation (hence I'll rename the title). There's some very basic support to validate the graph structure and layout options themselves. That's a different topic though and not active in most use cases.

**Comment by Baltoli** [comment]
This is exactly what I was looking for - thank you!
